From 0351a07fe5fe3c82cc8bdc4d85de9ff4624945c6 Mon Sep 17 00:00:00 2001
From: Peter Robinson <pbrobinson@gmail.com>
Date: Wed, 16 Oct 2019 10:06:33 +0100
Subject: [PATCH] Revert "rpmrc: Add architecture compatibility mapping between
 aarch64 and arm64"

This reverts commit 236d6f5a2b924266b1249a82875b595e8758c52b.

This change is fundamentally wrong. arm64 is a catch all that covers the
naming but rpm uses "uname -m" to detect the actual architecture. The aarch64
nonclamenture specifies the ISA (Instruction Set Architecture) for the 64-bit
variant of the Arm architecture which was first supported with ARMv8.

Because all 64-bit variants of ARMv8 use aarch64 the arm64 would never be
used in rpm. If it's useful in some context while using some Debian related
tools those tools should have some translation added there. This just adds
confusion and causes issues and duplication that haven't been necessary in
any of the arch64 work up until now.

Signed-off-by: Peter Robinson <pbrobinson@gmail.com>
---
 rpmrc.in | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/rpmrc.in b/rpmrc.in
index a3ad17a3a..cc212d4ba 100644
--- a/rpmrc.in
+++ b/rpmrc.in
@@ -103,7 +103,6 @@ optflags: sh4 -O2 -g -mieee
 optflags: sh4a -O2 -g -mieee
 
 optflags: aarch64 -O2 -g
-optflags: arm64 -O2 -g
 
 optflags: riscv64 -O2 -g
 
@@ -252,9 +251,7 @@ arch_canon:	sh3: sh3	17
 arch_canon:	sh4: sh4	17
 arch_canon:	sh4a: sh4a	17
 arch_canon:	xtensa: xtensa	18
-
-arch_canon:	aarch64: aarch64	19
-arch_canon:	arm64: arm64		19
+arch_canon:	aarch64: aarch64 19
 
 arch_canon:	mipsr6: mipsr6	20
 arch_canon:	mipsr6el: mipsr6el	20
@@ -393,7 +390,6 @@ buildarchtranslate: sh4: sh4
 buildarchtranslate: sh4a: sh4
 
 buildarchtranslate: aarch64: aarch64
-buildarchtranslate: arm64: aarch64
 
 buildarchtranslate: riscv: riscv64
 buildarchtranslate: riscv64: riscv64
@@ -507,8 +503,7 @@ arch_compat: sh3: noarch
 arch_compat: sh4: noarch
 arch_compat: sh4a: sh4
 
-arch_compat: aarch64: arm64 noarch
-arch_compat: arm64: aarch64 noarch
+arch_compat: aarch64: noarch
 
 arch_compat: riscv: noarch
 arch_compat: riscv64: noarch
@@ -545,7 +540,6 @@ os_compat: Darwin: MacOSX
 buildarch_compat: ia64: noarch
 
 buildarch_compat: aarch64: noarch
-buildarch_compat: arm64: aarch64
 
 buildarch_compat: riscv: noarch
 buildarch_compat: riscv64: noarch
